core/src/main/java/com/sequenceiq/cloudbreak/service/upgrade/image/filter/ImageFilterUpgradeService.java

```diff
@@ -0,0 +1,33 @@
+package com.sequenceiq.cloudbreak.service.upgrade.image.filter;
+
+import java.util.List;
+import java.util.Set;
+
+import javax.inject.Inject;
+
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.cloudbreak.cloud.model.catalog.Image;
+import com.sequenceiq.cloudbreak.service.upgrade.image.ImageFilterParams;
+import com.sequenceiq.cloudbreak.service.upgrade.image.ImageFilterResult;
+
+@Service
+public class ImageFilterUpgradeService {
+
+    public static final String EMPTY_REASON = "";
+
+    @Inject
+    private Set<UpgradeImageFilter> upgradeImageFilters;
+
+    public ImageFilterResult filterImages(List<Image> availableImages, ImageFilterParams imageFilterParams) {
+        ImageFilterResult result = new ImageFilterResult(availableImages, EMPTY_REASON);
+        for (UpgradeImageFilter imageFilter : upgradeImageFilters) {
```

Comment:
I have some concerns with this solution:
- it seems to me the filtering order is depending on Spring injection order, which could be different for every startup of the service, at least theoretically, which means 2 filtering could have different results depending which is the last filter. I think the filters should have some ordering, as we had previously.
- we still won't have any idea at least about which images made to the last round of filtering if no images available for upgrade. Most of our questions from customer are like: `Why the upgrade for version X is not possible` I think this refactor should also focus on helping us to answer this question.